src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java

```diff
@@ -26,10 +26,9 @@
     public ResponseEntity create(@Valid @RequestBody SmallFull smallFull,
             HttpServletRequest request)
             throws NoSuchAlgorithmException {
-        Transaction transaction = (Transaction) request.getSession()
-                .getAttribute(AttributeName.TRANSACTION.getValue());
-
-        SmallFull result = smallFullService.save(smallFull, transaction.getCompanyNumber());
+        CompanyAccountEntity companyAccountEntity = (CompanyAccountEntity) request.getSession()
+                .getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
+        SmallFull result = smallFullService.save(smallFull, companyAccountEntity.getId());
```

Comment:
Same as previous comment - we should use the 'ResponseObject' pattern for the service to check for errors and respond accordingly with the correct code in the controller